src/utils/gaa.js

```diff
@@ -1735,29 +1744,39 @@ export class GaaMetering {
         // This is only relevant for publishers doing SwG
         handleSwGEntitlement();
       } else if (
-        !GaaMetering.isUserRegistered(GaaMetering.userState) &&
+        !GaaMetering.isUserRegistered() &&
         GaaMetering.isGaa(allowedReferrers)
       ) {
         // This is an anonymous user so show the Google registration intervention
         showGoogleRegwall();
       } else {
         // User does not any access from publisher or Google so show the standard paywall
         callSwg((subscriptions) => {
-          subscriptions.setShowcaseEntitlement({
-            entitlement: ShowcaseEvent.EVENT_SHOWCASE_NO_ENTITLEMENTS_PAYWALL,
-            isUserRegistered: GaaMetering.isUserRegistered(
-              GaaMetering.userState
-            ),
-          });
+          switch (GaaMetering.userState.paywallReason) {
+            case 'RESERVED_USER':
+              subscriptions.setShowcaseEntitlement({
+                entitlement: ShowcaseEvent.EVENT_SHOWCASE_INELIGIBLE_PAYWALL,
+                isUserRegistered: GaaMetering.isUserRegistered(),
+              });
+              break;
+            default:
+              subscriptions.setShowcaseEntitlement({
+                entitlement:
+                  ShowcaseEvent.EVENT_SHOWCASE_NO_ENTITLEMENTS_PAYWALL,
+                isUserRegistered: GaaMetering.isUserRegistered(),
+              });
+          }
         });
         // Show the paywall
         showPaywall();
       }
     });
   }
 
-  static isUserRegistered(userState) {
-    return userState.id !== undefined && userState.id != '';
+  static isUserRegistered() {
```

Comment:
This isn't a backwards compatible change. Maybe it should be something like

```
static isUserRegistered(userState) {
    if (userState) {
        return userState.id !== undefined && userState.id != '';
    }

  return (
      GaaMetering.userState.id !== undefined && GaaMetering.userState.id != ''
    );
}
```

Reply:
Added a new function: `isCurrentUserRegistered`. Having said that, nobody should be using `isUserRegistered` so it should be safe to make this change